src/hotspot/share/opto/intrinsicnode.hpp

```diff
@@ -166,6 +166,46 @@ class CountPositivesNode: public StrIntrinsicNode {
   virtual const Type* bottom_type() const { return TypeInt::POS; }
 };
 
+//------------------------------VectorizedHashCodeNode----------------------
+class VectorizedHashCodeNode: public Node {
+ public:
+  // Possible hashing modes, corresponding to the primitive being encoded,
+  // as well as adjusting for special treatment of various encoding of String
+  // arrays. Must correspond to declared constants in jdk.internal.util.ArraysSupport
+  typedef enum HashModes { LATIN1 = 0, UTF16 = 1, BYTE = 2, CHAR = 3, SHORT = 4, INT = 5 } HashMode;
```

Comment:
UTF16 can easily be replaced with CHAR by lifting up the shift as you say, but LATIN1 needs to be distinguished from BYTE since the former needs unsigned semantics. Modeling in a signed/unsigned input is possible, but I figured we might as well call it UNSIGNED_BYTE and decouple it logically from String::LATIN1.

Reply:
FTR `T_BOOLEAN` effectively represents unsigned byte.